Source/FullScreenPokemon.ts

```diff
@@ -6667,6 +6667,81 @@ module FullScreenPokemon {
                     }));
         }
 
+        /**
+         * Cutscene for a Scratch attack in battle.
+         * 
+         * @param FSP
+         * @param settings   Settings used for the cutscene.
+         * @param args   Settings for the routine.
+         */
+        cutsceneBattleAttackScratch(FSP: FullScreenPokemon, settings: IBattleCutsceneSettings, args: IBattleAttackRoutineSettings): void {
+            let defenderName: string = args.defenderName;
+            let defender: IThing = <IThing>FSP.BattleMover.getThing(defenderName);
+            let dt: number = 1;
+            let direction: number = defenderName === "opponent" ? -1 : 1;
+            let scratchLine: string = "ScratchLine";
+            let startX: number;
+            let startY: number;
+            let differenceX: number = defender.width / 2 * FSP.unitsize;
+            let lineArray: IThing[] = [];
+            let menu: IMenu = <IMenu>FSP.MenuGrapher.getMenu("BattleDisplayInitial");
+            let scratches: IThing[] = [
+                FSP.ObjectMaker.make("ExplosionSmall"),
+                FSP.ObjectMaker.make("ExplosionSmall"),
+                FSP.ObjectMaker.make("ExplosionSmall")
+            ];
+
+            if (direction === -1) {
+                startX = menu.right - defender.width / 2 * FSP.unitsize;
+                startY = menu.top;
+            } else {
+                startX = menu.left + defender.width * FSP.unitsize;
+                startY = menu.bottom - (defender.height + 8) * FSP.unitsize;
+            }
+
+            FSP.addThing(scratches[0], startX, startY);
+            let offset: number = scratches[0].width * FSP.unitsize / 2;
+            FSP.addThing(scratches[1], startX + offset * direction * -1, startY + offset);
+            FSP.addThing(scratches[2], startX + offset * direction * -2, startY + offset * 2);
+
+            FSP.TimeHandler.addEventInterval(
+                function (): void {
+                    for (let i: number = 0; i < 3; i += 1) {
+                        let line: IThing;
```

Comment:
Nit: move `let line` to where it's actually created, so `line` has as little scope as possible.